click/core.py

```diff
@@ -315,6 +315,12 @@ def __init__(self, command, parent=None, info_name=None, obj=None,
         #: will do its best to not cause any failures.
         self.resilient_parsing = resilient_parsing
 
+        #: Indicates that default values should be ignored.
+        #: Useful for completion.
+
+        #: .. versionadded:: 7.0
+        self.ignore_default_values = ignore_default_values
```

Comment:
This doesn't seem like an option that should be exposed to users. Is there a better way to handle this only for completions without making it something users can mess with easily?

Reply:
The only other thing that comes to mind would be when we check that we are in completion mode based on the presence of the completion environment variable we could set a module local variable in core.py. These doesn't seem to be a precedence for that though. Let me know if you had other ideas and what you prefer.

Reply:
If this is the only sticking point for this PR, what about using `resilient_parsing` mode for this purpose instead? It would be a subtle change in what `resilient_parsing` does, but the whole point of that flag is to turn off a bunch of normal parsing behaviors to make completion work.

Is there a use for setting `resilient_parsing=True` and ensuring that parameters are populated with their defaults? I can't imagine one.

Reply:
Good idea, let's roll this into `resilient_parsing`. Just be sure to document the change in behavior.